frontend/typed_args_test.py

```diff
@@ -28,16 +40,28 @@ def testReaderPosArgs(self):
             value.Int(0xbeef),
             value.Str('bar'),
         ]
-        reader = typed_args.Reader(list(pos_args), {})
+        reader = typed_args.Reader(list(pos_args), {}, arg_list, False)
 
-        # Haven't processed any args yet...
-        self.assertRaises(error.TypeErrVerbose, reader.Done)
+        # Haven't all the args
+        with self.assertRaises(error.TypeErrVerbose) as cm:
+            _ = reader.PosInt()
+            _ = reader.PosStr()
+            reader.Done()
+
+        # Check that the error points to the first unconsummed argument
+        e = cm.exception
+        self.assertEqual(pos_exprs[2].c, e.location)
 
         # Arg is wrong type...
-        self.assertRaises(error.TypeErrVerbose, reader.PosStr)
+        with self.assertRaises(error.TypeErrVerbose) as cm:
```

Comment:
hm didn't know you could do this, seems nicer than the old style

Reply:
yeah, it's pretty handy